### ISSUE TITLE
Missing check on COLLECT

### ISSUE THREAD

**Issue by suynwa** [SAP/code-pal-for-abap#441]
> 💡 Code pal for ABAP follows the [Clean ABAP](https://github.com/SAP/styleguides/blob/main/clean-abap/CleanABAP.md). If the issue relates to coding style, please submit it [here](https://github.com/SAP/styleguides/issues). 

**Check Name**
> Missing check on COLLECT

**Actual Behavior**
Not relevant

**Expected Behavior**
I could not find any check in the standard ATC w.r.t usage of the COLLECT statement. As per the [standard SAP programming guideline ](https://help.sap.com/doc/abapdocu_755_index_htm/7.55/en-US/abencollect_guidl.htm), COLLECT should not be used on standard tables anymore. 

Do you think it makes sense to implement such a check in Code-Pal?

**Comment by lucasborin** [comment]
@suynwa : I created a new check under the `new_check_collect` branch. Would you mind testing it?

**Comment by lucasborin** [comment]
Known GAPs:
- `COLLECT` in Internal tables typed as table types (DDIC) are not being considered

**Comment by lucasborin** [comment]
Should we apply this rule if the internal table is typed as `RANGE OF ...`?

**Comment by suynwa** [comment]
> Should we apply this rule if the internal table is typed as RANGE OF ...?

Why would someone want to use COLLECT on ranges🤦‍♂️? Maybe to make the check fool-proof we can extend this rule for ranges. 

Btw, aren't ranges special type of standard tables? What info does RTTI return for ranges?

**Comment by lucasborin** [comment]
Yes, it is. The [documentation](https://help.sap.com/doc/abapdocu_752_index_htm/7.52/en-US/abaptypes_ranges.htm) says:
> A ranges table is a standard table with a standard key

So, I bet the `COLLECT` on `RANGES` is not valid anymore. 

> What info does RTTI return for ranges?
```abap
TYPES: BEGIN OF linetype,
         sign   TYPE c LENGTH 1,
         option TYPE c LENGTH 2,
         low    {TYPE type}|{LIKE dobj},
         high   {TYPE type}|{LIKE dobj},
       END OF linetype.
```

**Comment by lucasborin** [comment]
How about this?

```abap
  DATA: BEGIN OF lt_bukrs OCCURS 0,
          bukrs         LIKE j_1bnfdoc-bukrs,
        END OF lt_bukrs,

  LOOP AT ti_list.
    MOVE ti_list-bukrs TO lt_bukrs.
    COLLECT lt_bukrs.
  ENDLOOP.
```
